src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -2345,24 +2349,27 @@ class TransactionStatementSegment(BaseSegment):
 
     type = "transaction_statement"
     match_grammar = OneOf(
-        # BEGIN | SAVE TRANSACTION
-        # COMMIT [ TRANSACTION | WORK ]
-        # ROLLBACK [ TRANSACTION | WORK ]
+        # [ BEGIN | SAVE ] [ TRANSACTION | TRAN ]
+        # COMMIT [ TRANSACTION | TRAN | WORK ]
+        # ROLLBACK [ TRANSACTION | TRAN | WORK ]
         # https://docs.microsoft.com/en-us/sql/t-sql/language-elements/begin-transaction-transact-sql?view=sql-server-ver15
         Sequence(
             "BEGIN",
             Sequence("DISTRIBUTED", optional=True),
-            "TRANSACTION",
+            Ref("TransactionGrammar"),
             Ref("SingleIdentifierGrammar", optional=True),
             Sequence("WITH", "MARK", Ref("QuotedIdentifierSegment"), optional=True),
             Ref("DelimiterSegment", optional=True),
         ),
         Sequence(
             OneOf("COMMIT", "ROLLBACK"),
-            OneOf("TRANSACTION", "WORK", optional=True),
+            OneOf(Ref("TransactionGrammar"), "WORK", optional=True),
             Ref("DelimiterSegment", optional=True),
         ),
-        Sequence("SAVE", "TRANSACTION", Ref("DelimiterSegment", optional=True)),
+        Sequence(
+            "SAVE",
+            Ref("TransactionGrammar"),
+            Ref("DelimiterSegment", optional=True)),
```

Comment:
Could you also include a unit test example for `SAVE` as well please?

Reply:
Note this PR didn't add `SAVE`. It's only in green as a reformating.

But, saying that, additional test cases always appreciated!

Reply:
good point, I double checked and there's no existing test case for `SAVE` so would be nice to add to the test case for completeness 😄

Reply:
If by unit test you mean the `sql`/`yaml` files, I've added a very simple one.